### ISSUE TITLE
No results found in case if suggestions.length equals to limit

### ISSUE THREAD

**Issue by abguy** [corejavascript/typeahead.js#30]
Hello,

We use typeahead.js version '0.11.1'.

Our code looks like:

``` js
this.bloodhound = new Bloodhound({
    datumTokenizer: Bloodhound.tokenizers.obj.whitespace('name'),
    queryTokenizer: Bloodhound.tokenizers.whitespace,
    identify:       this.identify.bind(this),
    local:          this.localData,
    remote: {
        url:        this.searchUrl,
        wildcard:   this.searchWildcard
    }
});

$(this.input).typeahead({
    highlight:  true,
    hint:       true,
    minLength:  1
}, {
    display:    this.toString.bind(this),
    source:     this.bloodhound,
    limit:      5,
    templates: {
        pending:    this.loadingTemplate,
        empty:      this.emptyTemplate
    }
});
```

As you can see we have no prefetch/cache enabled, but we have some local data before initialization.
## Problem

If I type something and exactly 5 items are returned I see "No items found" message (this.emptyTemplate). But there were 5 items found!
## Additional info

I think that this issue is caused by Dataset.update() function logic:

``` js
rendered += suggestions.length;
that._append(query, suggestions.slice(0, that.limit - rendered));
```

Thus it appends zero items.

BTW, typeahead is awesome! Thanks!

**Comment by mlissner** [comment]
I think I'm seeing this too. it's enough to make you feel absolutely crazy. Did you find a workaround? Frankly, I don't want any limit logic at all, since my remote endpoint just returns 20 or fewer results no matter what.

**Comment by jvdbrink20** [comment]
I use the limit: Infinity workaround suggested in the other github repo. I limit my results server side.
Couldn't find the 'limit' property usage in code so I can't suggest a fix and do a PR.

**Comment by jlbooker** [comment]
Looks like this was fixed in #52, with further discussion in #43. Closing.
